### PR TITLE
BIGTOP-4105: Build fails when '-DpkgSuffix' is specified and '(rpm|deb)_pkg_suffix' is not specified in bigtop.bom

### PR DESCRIPTION
**Pull request by Kapkiai**
### Description of PR
Build fails when '-DpkgSuffix' is specified when '(rpm|deb)_pkg_suffix'  is not specified in bigtop.bom
[Issue](https://issues.apache.org/jira/browse/BIGTOP-4088)/https://github.com/apache/bigtop/pull/1254 introduced this bug by not checking whether the '(rpm|deb)_pkg_suffix' has been specified in bigtop.bom.
When config '(rpm|deb)_pkg_suffix'  is not specified, groovy returns an empty map  '[:]' which breaks the build. 

``` bash
./gradlew clean bigtop-groovy-pkg bigtop-jsvc-pkg bigtop-select-pkg bigtop-utils-pkg  -Dbuildwithdeps=true -PparentDir=/usr/bigtop -PpkgSuffix
........
Task :bigtop-groovy-srpm
Wrote: /home/mathew/Projects/bigtop/build/bigtop-groovy/rpm/SRPMS/bigtop-groovy-2.5.4-1.el8.src.rpm> Task :bigtop-utils-srpm
Wrote: /home/mathew/Projects/bigtop/build/bigtop-utils/rpm/SRPMS/bigtop-utils-3.3.0-1.el8.src.rpm> Task :bigtop-utils-rpm FAILED
error: cannot open /home/mathew/Projects/bigtop/output/bigtop-utils/bigtop-utils[:]-3.3.0-1.el8.src.rpm: No such file or directoryFAILURE: Build failed with an exception.
```

Sample code to illustrate the source of bug. Below groovy sample is how packages.gradle access project config in bigtop.bom

``` groovy
def bom = """bigtop {
    name = "ambari"
}"""

def conf = new ConfigSlurper().parse(bom)

println('Output')
println(conf.bigtop.name)
println(conf.bigtop.not_exist) 

// Output
// ambari
// [:]
```
### How was this patch tested?
``` bash
./gradlew clean bigtop-groovy-pkg bigtop-jsvc-pkg bigtop-select-pkg bigtop-utils-pkg  -Dbuildwithdeps=true -PparentDir=/usr/bigtop -PpkgSuffix

Executing(--clean): /bin/sh -e /var/tmp/rpm-tmp.ZXB60E
+ umask 022
+ cd /home/mathew/Projects/bigtop/build/bigtop-select/rpm//BUILD
+ rm -rf bigtop-select-3.3.0
+ exit 0

BUILD SUCCESSFUL in 8s
48 actionable tasks: 24 executed, 24 up-to-date

```

### For code changes:

- [ ] Does the title or this PR starts with the corresponding JIRA issue id (e.g. 'BIGTOP-3638. Your PR title ...')?
- [ ] Make sure that newly added files do not have any licensing issues. When in doubt refer to https://www.apache.org/licenses/

### CODE REVIEW

**Comment by Kapkiai** [comment]
@guyuqi @iwasakims @sekikn
Could you help review this PR?

File: packages.gradle [hunk 1]
```diff
@@ -102,7 +102,7 @@ def isValidMavenBuildThreads(threads) {
 }
 
 def getPkgNameSuffix(component, type) {
-    def originalSuffix = type.equalsIgnoreCase("deb") ? component.deb_pkg_suffix : component.rpm_pkg_suffix
+    def originalSuffix = type.equalsIgnoreCase("deb") ? component.deb_pkg_suffix?:"" : component.rpm_pkg_suffix?:"%{nil}"
```
Comment (iwasakims) [inline packages.gradle:105]: @Kapkiai I feel chaining ternary operator and elvis operator spoils readability. How about splitting the line and adding whitespace around the `?:` operator? like:

```
type.equalsIgnoreCase("deb") ?
    component.deb_pkg_suffix ?: "" :
    component.rpm_pkg_suffix ?: "%{nil}"
```

Comment (Kapkiai) [reply packages.gradle:105]: @iwasakims good suggestion. I have made changes.

Comment (Kapkiai) [reply packages.gradle]: @iwasakims is it a good path to refactor the changes as below?

``` groovy
def defaultSuffix = type.equalsIgnoreCase("deb") ?  "" : "%{nil}"
def originalSuffix = type.equalsIgnoreCase("deb") ?
            component.deb_pkg_suffix ?: defaultSuffix :
            component.rpm_pkg_suffix ?: defaultSuffix
```

With that, we will have the defaults defined once.

Comment (iwasakims) [reply packages.gradle]: I think we should avoid extra variable definition if it is used only here.

**Review by iwasakims — APPROVED** [review]
LGTM. Thanks for the contribution , @Kapkiai.

**Pull request merged**